src/main/scala/flow.scala

```diff
@@ -8,9 +8,36 @@ import scala.language.experimental.macros
 @compileTimeOnly("implementation detail of flow comprehensions. don't use yourself")
 final class FlowContext private()
 
+abstract class Constructor[H[_]]{
+  def create[T](v: T): H[T]
+}
```

Comment:
How about a macro that makes instances of this in terms of companion `apply` and just a few instances defined with higher priority for the few cases where we need them?